src/hooks/useIsolatedI18n.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License as published by the Free Software Foundation, version 3 of the License.
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+ * Other licensing options may be available, please reach out to data-viz@feedzai.com for more information.
+ */
+
+import { useEffect, useMemo } from "react";
+import { createInstance, i18n } from "i18next";
+import { initReactI18next, useTranslation } from "react-i18next";
+// @ts-expect-error - i18n.js file doesn't have TypeScript declarations
+import globalI18n from "../i18n";
+
+export type TranslationObject = {
+	[key: string]: string;
+};
+
+export type CustomTranslations = {
+	[languageCode: string]: TranslationObject;
+};
```

Comment:
I would delete `TranslationObject` and use the "native" `Record<string, string>` instead.